test/python/test_reactor.py

```diff
@@ -3110,3 +3122,153 @@ def deltaC():
         assert r1.thermo.P == approx(151561.15, rel=1e-6)
         assert r1.thermo["H2"].Y[0] == approx(0.13765976, rel=1e-6)
         assert r2.thermo["O2"].Y[0] == approx(0.94617029, rel=1e-6)
+
+
+class TestSteadySolver:
+    @pytest.mark.parametrize("reactor_class",
+        [ct.Reactor, ct.IdealGasReactor,
+         ct.MoleReactor, ct.IdealGasMoleReactor])
+    def test_const_volume(self, reactor_class):
+        gas = ct.Solution("h2o2.yaml", transport_model=None)
+        gas.set_equivalence_ratio(1.2, "H2:1.0", "O2:1.0, N2:3.76")
+        gas.TP = 500, 20 * ct.one_atm
+
+        upstream = ct.Reservoir(gas)
+        gas.equilibrate("HP")
+        downstream = ct.Reservoir(gas)
+        V0 = 1e-3
+        r = reactor_class(gas, volume=V0)
+        inlet = ct.MassFlowController(upstream, r, mdot=120)
+        ct.PressureController(r, downstream, primary=inlet)
+        net = ct.ReactorNet([r])
+        net.solve_steady()
+        assert r.volume == approx(V0)
+        assert r.thermo.T == approx(2429.2709)
+        assert r.mass == approx(0.002288176)
+
+    @pytest.mark.parametrize("reactor_class",
+        [ct.ConstPressureReactor, ct.IdealGasConstPressureReactor])
+    def test_const_pressure(self, reactor_class):
+        gas = ct.Solution("h2o2.yaml", transport_model=None)
+        gas.set_equivalence_ratio(1.2, "H2:1.0", "O2:1.0, N2:3.76")
+        gas.TP = 500, 20 * ct.one_atm
+
+        upstream = ct.Reservoir(gas)
+        gas.equilibrate("HP")
+        downstream = ct.Reservoir(gas)
+        r = reactor_class(gas, volume=1e-3)
+        m0 = r.mass
+        ct.MassFlowController(upstream, r, mdot=160)
+        ct.MassFlowController(r, downstream, mdot=160)
+        net = ct.ReactorNet([r])
+        net.solve_steady()
+
+        assert r.mass == approx(m0)
+        assert r.thermo.T == approx(2407.35011)
+
+    @pytest.mark.parametrize("reactor_class",
+        [ct.IdealGasReactor, ct.IdealGasMoleReactor])
```

Comment:
Is it worth adding some tests of the disallowed cases? On one hand it's more overhead, on the other it'd be straightforward to add and remove. :shrug:

Reply:
I've added some (specifically, the ones with reactor surfaces), but I think doing all the cases would be a bit too much.